File: domain/src/main/java/com/quhxuxm/quh/project/simpleblog/domain/ArticleComment.java
package com.quhxuxm.quh.project.simpleblog.domain;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

@Entity
@Table(name = "article_comment")
public class ArticleComment implements Serializable {
    private static final long serialVersionUID = -8397645527218208099L;
    @Id
    @GeneratedValue
    @Column(name = "id")
    private Long id;
    @ManyToOne
    @JoinColumn(name = "author_id", nullable = false, updatable = false)
    private Author author;
    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "create_date", nullable = false, updatable = false)
    private Date createDate;
    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "update_date", nullable = false, updatable = false)
    private Date updateDate;
    @Column(name = "content", length = 2000, nullable = false)
    private String content;
    @ManyToOne
    @JoinColumn(name = "parent_id", nullable = false, updatable = false)
    private ArticleComment parent;
    @ManyToOne(targetEntity = Article.class)
    @JoinColumn(name = "article_id", nullable = false, updatable = false)
    private Article article;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Author getAuthor() {
        return author;
    }

    public void setAuthor(Author author) {
        this.author = author;
    }

    public Date getCreateDate() {
        return createDate;
    }

    public void setCreateDate(Date createDate) {
        this.createDate = createDate;
    }

    public Date getUpdateDate() {
        return updateDate;
    }

    public void setUpdateDate(Date updateDate) {
        this.updateDate = updateDate;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public Article getArticle() {
        return article;
    }

    public void setArticle(Article article) {
        this.article = article;
    }

    public ArticleComment getParent() {
        return parent;
    }

    public void setParent(ArticleComment parent) {
        this.parent = parent;
    }
}

File: client/src/app/vo/article-comment.ts
export class ArticleComment {
  private _authorId: number;
  private _authorNickname: string;
  private _authorIconId: number;
  private _createDate: Date;
  private _updateDate: Date;
  private _content: string;
  private _parent: ArticleComment;

  get authorId(): number {
    return this._authorId;
  }

  set authorId(value: number) {
    this._authorId = value;
  }

  get authorNickname(): string {
    return this._authorNickname;
  }

  set authorNickname(value: string) {
    this._authorNickname = value;
  }

  get authorIconId(): number {
    return this._authorIconId;
  }

  set authorIconId(value: number) {
    this._authorIconId = value;
  }

  get createDate(): Date {
    return this._createDate;
  }

  set createDate(value: Date) {
    this._createDate = value;
  }

  get updateDate(): Date {
    return this._updateDate;
  }

  set updateDate(value: Date) {
    this._updateDate = value;
  }

  get content(): string {
    return this._content;
  }

  set content(value: string) {
    this._content = value;
  }

  get parent(): ArticleComment {
    return this._parent;
  }

  set parent(value: ArticleComment) {
    this._parent = value;
  }
}
